### PR TITLE
Add Alert related docs

### PR DESCRIPTION
**Pull request by zhishending**
Roadmap workitem ID:13682

### CODE REVIEW

**Review by RuiHou105 — CHANGES_REQUESTED** [review]
@zhishending  could you please 
1.update the changelog file
2.follow this [guidance ](https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211772/Content-review-process-summary)to create a roadmap work item and attach it in the PR description. Here is an [example](https://microsoftgraph.visualstudio.com/Roadmap/_workitems/edit/11551/).

File: api-reference/beta/resources/devicemanagement-alertrecord.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+|[getPortalNotifications](../api/devicemanagement-alertrecord-getportalnotifications.md)|[microsoft.graph.deviceManagement.portalNotification](../resources/devicemanagement-portalnotification.md) collection|View a list of all portal notifications that are ready to be consumed for current user. The portal notifications can be used to publish MEM portal notifications.|
+|[setPortalNotificationAsSent](../api/devicemanagement-alertrecord-setportalnotificationassent.md)|None|Set a single portal notification status to published. Will set isPortalNotificationSent property to true for current user.|
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|alertImpact|[microsoft.graph.deviceManagement.alertImpact](../resources/devicemanagement-alertimpact.md)|The impact of the alert event. Consists of a number followed by the aggregation type. For example: 6 Count, 12 AffectedCloudPcPercentage, etc|
+|alertRuleId|String|The corresponding alert rule id.|
+|alertRuleTemplate|alertRuleTemplate|The rule template of the alert event.The possible values are: `cloudPcProvisionScenario`, `cloudPcImageUploadScenario`, `cloudPcOnPremiseNetworkConnectionCheckScenario`, `unknownFutureValue`.|
```
Comment (RuiHou105) [inline api-reference/beta/resources/devicemanagement-alertrecord.md:36]: The type for property `alertRuleTemplate `is an enum, so please also show the enum alertRuleTemplate. You can refer to this [example ](https://github.com/microsoftgraph/microsoft-graph-docs/pull/15416/files#diff-4c47ba8875401e7419e011f294b6a03fcdb96833252320b84209a29ec58a30f6), cloudpcsnapshot.md file line 34.

File: api-reference/beta/resources/devicemanagement-alertrecord.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+|detectedDateTime|DateTimeOffset|The alert event detected time. The Timestamp type represents date and time information using ISO 8601 format. For example, midnight UTC on Jan 1, 2014 would look like this: '2014-01-01T00:00:00Z'.|
+|displayName|String|The display name of the alert record.|
+|id|String|The alert record id. Inherited from [entity](../resources/entity.md).|
+|lastUpdatedDateTime|DateTimeOffset|The last updated time of the alert record. The Timestamp type represents date and time information using ISO 8601 format. For example, midnight UTC on Jan 1, 2014 would look like this: '2014-01-01T00:00:00Z'.|
+|resolvedDateTime|DateTimeOffset|The resolved time if alert event is resolved. The Timestamp type represents date and time information using ISO 8601 format. For example, midnight UTC on Jan 1, 2014 would look like this: '2014-01-01T00:00:00Z'.|
+|severity|ruleSeverityType|The severity of the alert event. Possible values are informational, warning, critical, unknown and unknownFutureValue. The possible values are: `unknown`, `informational`, `warning`, `critical`, `unknownFutureValue`.|
```
Comment (RuiHou105) [inline api-reference/beta/resources/devicemanagement-alertrecord.md:42]: Same with alertRuleTemplate

File: api-reference/beta/resources/devicemanagement-alertrecord.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+|status|alertStatusType|The status of the alert record. Possible values are: active, resolved and unknownFutureValue .The possible values are: `active`, `resolved`, `unknownFutureValue`.|
```
Comment (RuiHou105) [inline api-reference/beta/resources/devicemanagement-alertrecord.md:40]: add the alertStatusType enum, following the example in previous comment.

File: api-reference/beta/resources/devicemanagement-alertrule.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+|[Create alertRule](../api/devicemanagement-monitoring-post-alertrules.md)|[microsoft.graph.deviceManagement.alertRule](../resources/devicemanagement-alertrule.md)|Create a new [alertRule](../resources/devicemanagement-alertrule.md) object.|
+|[Get alertRule](../api/devicemanagement-alertrule-get.md)|[microsoft.graph.deviceManagement.alertRule](../resources/devicemanagement-alertrule.md)|Read the properties and relationships of an [alertRule](../resources/devicemanagement-alertrule.md) object.|
+|[Update alertRule](../api/devicemanagement-alertrule-update.md)|[microsoft.graph.deviceManagement.alertRule](../resources/devicemanagement-alertrule.md)|Update the properties of an [alertRule](../resources/devicemanagement-alertrule.md) object.|
+|[Delete alertRule](../api/devicemanagement-alertrule-delete.md)|None|Deletes an [alertRule](../resources/devicemanagement-alertrule.md) object.|
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|alertRuleTemplate|alertRuleTemplate|The rule template of the rule.The possible values are: `cloudPcProvisionScenario`, `cloudPcImageUploadScenario`, `cloudPcOnPremiseNetworkConnectionCheckScenario`, `unknownFutureValue`.|
```
Comment (RuiHou105) [inline api-reference/beta/resources/devicemanagement-alertrule.md:32]: alertRuleTemplate is an enum. Please add the enumeration down below.

File: api-reference/beta/resources/devicemanagement-rulethreshold.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Includes information about an alert rule's threshold settings.
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|aggregation|aggregationType|Indicates the built-in aggregation methods. The possible values are: `count`, `percentage`, `affectedCloudPcCount`, `affectedCloudPcPercentage`, `unknownFutureValue`.|
+|operator|operatorType|Indicates the built-in operator. The possible values are: `greaterOrEqual`, `equal`, `greater`, `less`, `lessOrEqual`, `notEqual`, `unknownFutureValue`.|
```
Comment (RuiHou105) [inline api-reference/beta/resources/devicemanagement-rulethreshold.md:22]: Please add aggregationType enum and operatorType enum.

File: api-reference/beta/api/devicemanagement-alertrecord-get.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  "name": "get_alertrecord"
+}
+-->
+``` http
+GET https://graph.microsoft.com/beta/deviceManagement/monitoring/alertRecords/{alertRecordId}
+```
+
+
+### Response
+>**Note:** The response object shown here might be shortened for readability.
```
Comment (RuiHou105) [inline api-reference/beta/api/devicemanagement-alertrecord-get.md:64]: As you already showed the response object, you can remove this extra note in all files.

File: api-reference/beta/api/devicemanagement-alertrecord-getportalnotifications.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+}
+-->
+``` http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "value": [
+    {
+      "@odata.type": "microsoft.graph.deviceManagement.portalNotification"
```
Comment (RuiHou105) [inline api-reference/beta/api/devicemanagement-alertrecord-getportalnotifications.md:78]: Could you please add detailed response here?

File: api-reference/beta/api/devicemanagement-alertrule-update.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+### Request
+<!-- {
+  "blockType": "request",
+  "name": "update_alertrule"
+}
+-->
+``` http
+PATCH https://graph.microsoft.com/beta/deviceManagement/monitoring/alertRules/{alertRuleId}
+Content-Type: application/json
+Content-length: 457
```
Comment (RuiHou105) [inline api-reference/beta/api/devicemanagement-alertrule-update.md:69]: Why?

File: api-reference/beta/resources/devicemanagement-alertrule.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+doc_type: resourcePageType
+---
+
+# alertRule resource type
+
+Namespace: microsoft.graph.deviceManagement
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+**TODO: Add Description**
```
[the remark this answers is not in the document]
Comment (zhishending) [reply api-reference/beta/resources/devicemanagement-alertrule.md]: Updated in new commit.

File: api-reference/beta/resources/devicemanagement-alertrecord.md (hunk 1 above)
Comment (zhishending) [reply api-reference/beta/resources/devicemanagement-alertrecord.md]: Thanks for this information. Updated all enums in this PR.

File: api-reference/beta/resources/devicemanagement-alertrecord.md (hunk 2 above)
Comment (zhishending) [reply api-reference/beta/resources/devicemanagement-alertrecord.md]: Updated.

File: api-reference/beta/resources/devicemanagement-alertrecord.md (hunk 3 above)
Comment (zhishending) [reply api-reference/beta/resources/devicemanagement-alertrecord.md]: Updated.

File: api-reference/beta/api/devicemanagement-alertrecord-getportalnotifications.md (hunk 7 above)
Comment (zhishending) [reply api-reference/beta/api/devicemanagement-alertrecord-getportalnotifications.md:78]: The response contains an array of portalNotification objects. I think this is accurate. The details of microsoft.graph.deviceManagement.portalNotification is included in resource files. Could you please elabrate what we need to add here?

File: api-reference/beta/api/devicemanagement-alertrule-update.md (hunk 8 above)
Comment (zhishending) [reply api-reference/beta/api/devicemanagement-alertrule-update.md]: Deleted content length. Not sure why it was auto generated. Thanks for catching this.

File: api-reference/beta/api/devicemanagement-alertrecord-get.md (hunk 6 above)
Comment (zhishending) [reply api-reference/beta/api/devicemanagement-alertrecord-get.md:64]: Deleted all notes from API files.

File: api-reference/beta/api/devicemanagement-alertrecord-get.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Read the properties and relationships of an [alertRecord](../resources/devicemanagement-alertrecord.md) object.
+
+## Permissions
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
+
+|Permission type|Permissions (from least to most privileged)|
+|:---|:---|
+|Delegated (work or school account)|CloudPC.ReadWrite.All|
```
Comment (AshleyYangSZ) [inline api-reference/beta/api/devicemanagement-alertrecord-get.md:22]: Should be like this CloudPC.Read.All, CloudPC.ReadWrite.All

File: api-reference/beta/resources/devicemanagement-monitoring.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+The entry point entity type to access all alert related entities and APIs.
+
+Inherits from [entity](../resources/entity.md).
+
+## Methods
+|Method|Return type|Description|
+|:---|:---|:---|
+|[getEffectivePermissions](../api/devicemanagement-monitoring-geteffectivepermissions.md)|String collection|View the effective permissions of the currently authenticated user.|
+
```
Comment (AshleyYangSZ) [inline api-reference/beta/resources/devicemanagement-monitoring.md:24]: Remove this.

**Review by AshleyYangSZ — CHANGES_REQUESTED** [review]

File: api-reference/beta/api/devicemanagement-alertrecord-get.md (hunk 10 above)
Comment (zhishending) [reply api-reference/beta/api/devicemanagement-alertrecord-get.md]: Updated in new commit.

File: api-reference/beta/resources/devicemanagement-monitoring.md (hunk 11 above)
Comment (zhishending) [reply api-reference/beta/resources/devicemanagement-monitoring.md]: Removed in new commit.

File: api-reference/beta/resources/devicemanagement-alertrecord.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+# alertRecord resource type
+
+Namespace: microsoft.graph.deviceManagement
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+This entity represents the record of an alert event triggered by the rule. When an alert rule's threshold is reached, a record of the alert event will be generated and stored, and admins will receive notifications via defined notification channels.
+
+Inherits from [entity](../resources/entity.md).
```
Comment (AshleyYangSZ) [inline api-reference/beta/resources/devicemanagement-alertrecord.md:18]: Please remove this line.

File: api-reference/beta/toc.yml [hunk 13]
```diff
@@ -10229,6 +10229,39 @@ items:
...
+        - name: Alert recrod
+          href: resources/devicemanagement-alertrecord.md
+          items:
+          - name: List
+            href: api/devicemanagement-alertrecord-list.md
+          - name: Get
+            href: api/devicemanagement-alertrecord-get.md
+          - name: Get portal notifications
+            href: api/devicemanagement-alertrecord-getportalnotifications.md
+          - name: Set portal notification as sent
```
Comment (AshleyYangSZ) [inline api-reference/beta/toc.yml:10245]: Missed href of this action.

File: api-reference/beta/resources/devicemanagement-alertrule.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+# alertRule resource type
+
+Namespace: microsoft.graph.deviceManagement
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents the rules that IT Admin can config to trigger alerts.
+
+
+Inherits from [entity](../resources/entity.md).
```
Comment (AshleyYangSZ) [inline api-reference/beta/resources/devicemanagement-alertrule.md:19]: No need for this line.

File: api-reference/beta/toc.yml [hunk 15]
```diff
@@ -10229,6 +10229,39 @@ items:
...
+          items:
+          - name: List
+            href: api/devicemanagement-alertrule-list.md
+          - name: Get
+            href: api/devicemanagement-alertrule-get.md
+          - name: Create
+            href: api/devicemanagement-alertrule-post.md
+          - name: Update
+            href: api/devicemanagement-alertrule-update.md
+        - name: Alert impact
```
Comment (AshleyYangSZ) [inline api-reference/beta/toc.yml:10257]: What's your consideration of put these enum resources link here?

File: changelog/Microsoft.DeviceManagement.Monitoring.json [hunk 16]
```diff
@@ -0,0 +1,150 @@
+{
+    "changelog": [
+      {
+        "ChangeList": [
+          {
+            "Id": "ee70b5aa-c9c1-4951-99f8-9eeef35ceadc",
+            "ApiChange": "Resource",
+            "ChangedApiName": "deviceManagement",
+            "ChangeType": "Addition",
+            "Description": "Added the [deviceManagement](https://docs.microsoft.com/en-us/graph/api/resources/deviceManagement?view=graph-rest-beta) resource",
```
Comment (AshleyYangSZ) [inline changelog/Microsoft.DeviceManagement.Monitoring.json:10]: This changelog item seems not need.

File: changelog/Microsoft.DeviceManagement.Monitoring.json [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+            "Description": "Added the [setPortalNotificationAsSent](https://docs.microsoft.com/en-us/graph/api/setPortalNotificationAsSent?view=graph-rest-beta) method to the [setPortalNotificationAsSent](https://docs.microsoft.com/en-us/graph/api/setPortalNotificationAsSent?view=graph-rest-beta) resource.",
+            "Target": "setPortalNotificationAsSent"
+          },
+          {
+            "Id": "ee70b5aa-c9c1-4951-99f8-9eeef35ceadc",
+            "ApiChange": "Method",
+            "ChangedApiName": "getEffectivePermissions",
+            "ChangeType": "Addition",
+            "Description": "Added the [getEffectivePermissions](https://docs.microsoft.com/en-us/graph/api/getEffectivePermissions?view=graph-rest-beta) method to the [getEffectivePermissions](https://docs.microsoft.com/en-us/graph/api/getEffectivePermissions?view=graph-rest-beta) resource.",
+            "Target": "getEffectivePermissions"
```
Comment (AshleyYangSZ) [inline changelog/Microsoft.DeviceManagement.Monitoring.json:131]: We don't have this API currently. Could u help remove this?

File: api-reference/beta/resources/devicemanagement-alertrecord.md (hunk 12 above)
Comment (zhishending) [reply api-reference/beta/resources/devicemanagement-alertrecord.md]: Thanks for the great comments! Updated files according to all comments in new commit.

**Review by AshleyYangSZ — APPROVED** [review]

File: api-reference/beta/toc.yml [hunk 18]
```diff
@@ -10229,6 +10229,32 @@ items:
             href: api/cloudpcorganizationsettings-get.md
           - name: Update
             href: api/cloudpcorganizationsettings-update.md
+      - name: Monitoring (preview)
+        href: resources/devicemanagement-monitoring.md
+        displayName: Overview
+        items:
+        - name: Alert recrod
```
Comment (RuiHou105) [inline api-reference/beta/toc.yml:10236]: typo: recrod -> record

File: api-reference/beta/resources/devicemanagement-rulethreshold.md [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+``` json
+{
+  "@odata.type": "#microsoft.graph.deviceManagement.ruleThreshold",
+  "aggregation": "String",
+  "operator": "String",
+  "target": "Integer"
+}
+```
+
+<!--
```
Comment (RuiHou105) [inline api-reference/beta/resources/devicemanagement-rulethreshold.md:66]: Remove this unused part?

File: api-reference/beta/resources/devicemanagement-notificationchannel.md [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+  ]
+}
+```
+
+<!--
+{
+  "type": "#page.annotation",
+  "namespace": "microsoft.graph.deviceManagement"
+}
+-->
```
Comment (RuiHou105) [inline api-reference/beta/resources/devicemanagement-notificationchannel.md:59]: Remove the unused part?

File: api-reference/beta/resources/devicemanagement-alertrule.md [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+  ]
+}
+```
+
+<!--
+{
+  "type": "#page.annotation",
+  "namespace": "microsoft.graph.deviceManagement"
+}
+-->
```
Comment (RuiHou105) [inline api-reference/beta/resources/devicemanagement-alertrule.md:96]: Unused part?

File: api-reference/beta/resources/devicemanagement-alertrecord.md [hunk 22]
```diff
@@ -0,0 +1,84 @@
...
+  "lastUpdatedDateTime": "String (timestamp)"
+}
+```
+
+<!--
+{
+  "type": "#page.annotation",
+  "namespace": "microsoft.graph.deviceManagement"
+}
+-->
```
Comment (RuiHou105) [inline api-reference/beta/resources/devicemanagement-alertrecord.md:84]: What's this block line 79-84 used for?

**Review by RuiHou105 — CHANGES_REQUESTED** [review]
I left some minor comments. Please help resolve them. After this, you can add "Ready for content review " tag to trigger the review process.

File: api-reference/beta/toc.yml (hunk 18 above)
Comment (zhishending) [reply api-reference/beta/toc.yml]: Nice catch! Fixed in new commit.

**Review by RuiHou105 — APPROVED** [review]

**Pull request closed without merging**